src/parsing/buffers.rs

```diff
@@ -18,6 +18,28 @@ where
     Ok(n)
 }
 
+pub fn read_line_strict<R>(reader: &mut BufReader<R>, buf: &mut Vec<u8>, max_buf_len: u64) -> io::Result<usize>
```

Comment:
Can you add a test below for this function?

Reply:
Also maybe a test in this file as well https://github.com/sbstp/attohttpc/blob/master/src/parsing/response.rs#L21 to make sure it meshes well with an otherwise properly formatted response.

Reply:
Yes, I can do that. But I already found one issue with it: If the header contains CR that is not part of the line ending, the current code will fail, so I will probably have to refactor it considerably which will make it even more complicated. I am really not too enthused about it...

Reply:
Added tests here and there...